lib/compiler.js

```diff
@@ -670,51 +699,52 @@ function fixFileExtension(filepath, files, onlyScriptsExportingFunction) {
 
       if (!(otherFileExtension in getExcludedExtensions()) &&
         path.basename(f, otherFileExtension) == basename) {
-        if (!onlyScriptsExportingFunction)
-          results.push(otherFile);
-        else if (onlyScriptsExportingFunction &&
-          (typeof require.extensions[otherFileExtension]) === 'function') {
-          results.push(otherFile);
-        }
+        // if (!onlyScriptsExportingFunction)
+        results.push(otherFile);
+        //else if (onlyScriptsExportingFunction &&
+          //(typeof scriptExtensions[otherFileExtension]) === 'function') {
+          //results.push(otherFile);
+        //}
       }
     }
   });
   return (results.length > 0 ? results[0] : undefined);
 }
 
-function resolveAppScriptPath(rootDir, relativePath, resolveOptions) {
+function resolveAppScriptPath(rootDir, relativePath, resolveOptions,
+                              scriptExtensions) {
   var resolvedPath = resolveAppPath(rootDir, relativePath, resolveOptions);
   if (!resolvedPath) {
     return false;
   }
   var sourceDir = path.dirname(resolvedPath);
   var files = tryReadDir(sourceDir);
-  var fixedFile = fixFileExtension(resolvedPath, files, false);
+  var fixedFile = fixFileExtension(resolvedPath, files, false,
+                                   scriptExtensions);
   return (fixedFile === undefined ? resolvedPath : fixedFile);
 }
 
-function buildAllMixinInstructions(appRootDir, mixinDirs, mixinSources, options,
-  modelInstructions) {
-  var extensions = _.without(_.keys(require.extensions),
-    _.keys(getExcludedExtensions()));
-
+function buildAllMixinInstructions(appRootDir, mixins, mixinDirs, mixinSources,
+                                   scriptExtensions, normalization,
+                                   modelInstructions) {
   // load mixins from `options.mixins`
-  var sourceFiles = options.mixins || [];
```

Comment:
I tried to make code look consistent with [how `mixinDirs` and `mixinSources` are declared  and passed to the function.](https://github.com/supasate/loopback-boot/blob/8fef5f6fa8b0d2e37b1c38877805fbfeb656dbd6/lib/compiler.js#L112-L113).

Do you want to refactor those arguments by passing only `options` and set their default values in `buildAllMixinInstructions` instead?